src/um/propidl.rs

```diff
@@ -4,13 +4,390 @@
 // <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
 // All files in the project carrying such notice may not be copied, modified, or distributed
 // except according to those terms.
-use shared::minwindef::WORD;
-use shared::wtypes::VARTYPE;
+use shared::guiddef::{CLSID, FMTID, GUID, REFCLSID, REFFMTID};
+use shared::minwindef::{
+    BYTE, DWORD, FILETIME, FLOAT, HIBYTE, HIWORD, INT, LOBYTE, LOWORD, UINT, WORD
+};
+use shared::ntdef::{
+    BOOLEAN, CHAR, HRESULT, LARGE_INTEGER, LONG, LPSTR, LPWSTR, PVOID, SHORT,
+    UCHAR, ULARGE_INTEGER, ULONG, USHORT
+};
+use shared::wtypes::{
+    BSTR, BSTRBLOB, CLIPDATA, CY, DATE, DECIMAL, PROPID, VARIANT_BOOL, VARTYPE
+};
+use shared::wtypesbase::{BLOB, DOUBLE, LPOLESTR, SCODE};
+use um::oaidl::{IDispatch, LPSAFEARRAY};
+use um::objidlbase::IStream;
+use um::unknwnbase::{IUnknown, IUnknownVtbl};
+pub const PROPSETFLAG_DEFAULT: DWORD = 0;
+pub const PROPSETFLAG_NONSIMPLE: DWORD = 1;
+pub const PROPSETFLAG_ANSI: DWORD = 2;
+pub const PROPSETFLAG_UNBUFFERED: DWORD = 4;
+pub const PROPSET_BEHAVIOR_CASE_SENSITIVE: DWORD = 1;
+STRUCT!{struct VERSIONEDSTREAM {
+    guidVersion: GUID,
+    pStream: *mut IStream,
+}}
+pub type LPVERSIONEDSTREAM = *mut VERSIONEDSTREAM;
+macro_rules! TYPEDEF_CA {
+    ($name:ident, $t:ty) => { STRUCT!{struct $name {
```

Comment:
Your macro takes the parameters in the reverse order that it does in the original headers, nor do the parameter names match the original parameter names. `#define TYPEDEF_CA(type, name)`